### PR TITLE
Support signing by Keystone

### PR DESCRIPTION
**Pull request by renfengshi**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -7,6 +7,8 @@ prod
 .gitsecret/keys/random_seed
 !*.secret
 *.log
+.yalc
+yalc.lock
```
Comment (HeesungB) [inline .gitignore:11]: @renfengshi Why was the `yalc` file added here?

File: package.json [hunk 2]
```diff
@@ -66,6 +66,7 @@
     "@types/node": "^18.13.0",
     "@types/react": "^16.14.4",
     "@types/react-dom": "^16.9.11",
+    "@types/uuid": "^9.0.2",
```
Comment (HeesungB) [inline package.json:69]: @renfengshi Why was it added if it's not used anywhere else?

File: packages/extension/src/components/icon/camera.tsx [hunk 3]
```diff
@@ -0,0 +1,25 @@
+import React, { FunctionComponent } from "react";
+import { IconProps } from "./types";
+
+export const CameraIcon: FunctionComponent<IconProps> = ({
```
Comment (HeesungB) [inline packages/extension/src/components/icon/camera.tsx:4]: @renfengshi Could you look at another icon file and follow the convention for icon files?

File: packages/extension/src/components/icon/information-plain.tsx [hunk 4]
```diff
@@ -0,0 +1,29 @@
+import React, { FunctionComponent } from "react";
+import { IconProps } from "./types";
+
+export const InformationPlainIcon: FunctionComponent<IconProps> = ({
```
Comment (HeesungB) [inline packages/extension/src/components/icon/information-plain.tsx:4]: @renfengshi Could you look at another icon file and follow the convention for icon files?

File: packages/extension/src/components/icon/keystone-text.tsx [hunk 5]
```diff
@@ -0,0 +1,34 @@
+import React, { FunctionComponent } from "react";
+import { IconProps } from "./types";
+
+export const KeystoneTextIcon: FunctionComponent<IconProps> = ({
```
Comment (HeesungB) [inline packages/extension/src/components/icon/keystone-text.tsx:4]: @renfengshi Could you look at another icon file and follow the convention for icon files?

File: packages/extension/src/components/icon/keystone.tsx [hunk 6]
```diff
@@ -0,0 +1,28 @@
+import React, { FunctionComponent } from "react";
+import { IconProps } from "./types";
+
+export const KeystoneIcon: FunctionComponent<IconProps> = ({
```
Comment (HeesungB) [inline packages/extension/src/components/icon/keystone.tsx:4]: @renfengshi Could you look at another icon file and follow the convention for icon files?

File: packages/extension/src/components/keystone/error.tsx [hunk 7]
```diff
@@ -0,0 +1,64 @@
+import { FunctionComponent } from "react";
+import { Modal } from "../modal";
+import { Box } from "../box";
+import { Body2, Subtitle1 } from "../typography";
+import React from "react";
+import { ColorPalette } from "../../styles";
+import { Button } from "../button";
+
+export const KeystoneError: FunctionComponent<{
```
Comment (HeesungB) [inline packages/extension/src/components/keystone/error.tsx:9]: @renfengshi I think `KeystoneErrorModal` fits the intent a little better

File: packages/extension/src/pages/register/connect-keystone/index.tsx [hunk 8]
```diff
@@ -0,0 +1,141 @@
...
+
+export const ConnectKeystoneScene: FunctionComponent<{
+  name: string;
+  password: string;
+
+  // append mode일 경우 위의 name, password는 안쓰인다. 대충 빈 문자열 넣으면 된다.
+  appendModeInfo?: {
+    vaultId: string;
+    afterEnableChains: string[];
+  };
```
Comment (HeesungB) [inline packages/extension/src/pages/register/connect-keystone/index.tsx:22]: @renfengshi I think I'll delete the `appendModeInfo` parameter since it's not used in the current file.

File: packages/extension/src/pages/register/connect-keystone/scan.tsx [hunk 9]
```diff
@@ -0,0 +1,136 @@
...
+
+export const ScanKeystoneScene: FunctionComponent<{
+  name: string;
+  password: string;
+
+  // append mode일 경우 위의 name, password는 안쓰인다. 대충 빈 문자열 넣으면 된다.
+  appendModeInfo?: {
+    vaultId: string;
+    afterEnableChains: string[];
+  };
```
Comment (HeesungB) [inline packages/extension/src/pages/register/connect-keystone/scan.tsx:25]: @renfengshi I think I'll delete the `appendModeInfo` parameter since it's not used in the current file.

File: packages/extension/src/pages/register/enable-chains/index.tsx [hunk 10]
```diff
@@ -472,7 +472,7 @@ export const EnableChainsScene: FunctionComponent<{
 
         return new CoinPretty(
           chainStore.getChain(a.chainId).stakeCurrency,
-          "0"
+          "-1"
         );
       })();
```
Comment (HeesungB) [inline packages/extension/src/pages/register/enable-chains/index.tsx:477]: @renfengshi What's the intent of going from 0 to -1 here?

File: packages/extension/src/pages/register/enable-chains/index.tsx [hunk 11]
```diff
@@ -486,7 +486,7 @@ export const EnableChainsScene: FunctionComponent<{
 
         return new CoinPretty(
           chainStore.getChain(b.chainId).stakeCurrency,
-          "0"
+          "-1"
         );
       })();
```
Comment (HeesungB) [inline packages/extension/src/pages/register/enable-chains/index.tsx:491]: @renfengshi What's the intent of going from 0 to -1 here?

File: packages/extension/webpack.config.js [hunk 12]
```diff
@@ -131,6 +131,7 @@ module.exports = {
       crypto: require.resolve("crypto-browserify"),
       stream: require.resolve("stream-browserify"),
       process: require.resolve("process/browser"),
+      zlib: require.resolve("browserify-zlib"),
```
Comment (HeesungB) [inline packages/extension/webpack.config.js:134]: @renfengshi Why did you add `zlib`?

File: packages/extension/src/pages/sign/utils/handle-eth-sign.ts [hunk 13]
```diff
@@ -56,13 +76,61 @@ export const handleEthereumPreSign = async (
       }
 
       return connectAndSignEthWithLedger(
-        useWebHID,
+        (options as LedgerOptions).useWebHID,
```
Comment (HeesungB) [inline packages/extension/src/pages/sign/utils/handle-eth-sign.ts:79]: @renfengshi I think it would be better to split the function. rather than `(options as LedgerOptions)`

**Review by HeesungB — COMMENTED** [review]
@renfengshi 

I've left any questions in the code.
In addition, I have questions about the following.

- I only import the `MultiAccounts` interface from the `@keystonehq/keystone-sdk` library. I think it would be better to remove this part and make it an interface separately without installing the `@keystonehq/keystone-sdk` library.

- Please remove the information icon and interaction from this file `packages/extension/src/pages/register/connect-hardware/index.tsx`.

 - We need to apply the design to light mode.
 - Requires multilingual processing (Intl).

If you have any questions, please leave them here.
Thank you.

File: packages/extension/src/pages/register/enable-chains/index.tsx (hunk 10 above)
Comment (renfengshi) [reply packages/extension/src/pages/register/enable-chains/index.tsx:477]: To ensure the unsupported chains at bottom.

File: packages/extension/src/pages/register/enable-chains/index.tsx (hunk 11 above)
Comment (renfengshi) [reply packages/extension/src/pages/register/enable-chains/index.tsx:491]: To ensure the unsupported chains at bottom.

File: package.json (hunk 2 above)
Comment (renfengshi) [reply package.json:69]: Keystone signing request id is uuid. In development environment, typescript cannot get type of uuid.

File: packages/extension/src/components/icon/camera.tsx (hunk 3 above)
Comment (renfengshi) [reply packages/extension/src/components/icon/camera.tsx:4]: @HeesungB Sorry, I cannot get what you mean. Is the color which does not follow?

File: packages/extension/webpack.config.js (hunk 12 above)
Comment (renfengshi) [reply packages/extension/webpack.config.js:134]: @keystonehq/keystone-sdk required.

File: packages/extension/src/pages/register/enable-chains/index.tsx (hunk 11 above)
Comment (HeesungB) [reply packages/extension/src/pages/register/enable-chains/index.tsx:491]: @renfengshi I think you should keep this at 0 and do something different

File: packages/extension/src/pages/register/enable-chains/index.tsx (hunk 10 above)
Comment (HeesungB) [reply packages/extension/src/pages/register/enable-chains/index.tsx:477]: @renfengshi I think you should keep this at 0 and do something different

File: packages/extension/src/components/icon/camera.tsx (hunk 3 above)
Comment (renfengshi) [reply packages/extension/src/components/icon/camera.tsx:4]: @HeesungB What should I do to resolve this conversation?

**Comment by renfengshi** [comment]
> @renfengshi
> 
> I've left any questions in the code. In addition, I have questions about the following.
> 
> * I only import the `MultiAccounts` interface from the `@keystonehq/keystone-sdk` library. I think it would be better to remove this part and make it an interface separately without installing the `@keystonehq/keystone-sdk` library.
> * Please remove the information icon and interaction from this file `packages/extension/src/pages/register/connect-hardware/index.tsx`.
> * We need to apply the design to light mode.
> * Requires multilingual processing (Intl).
> 
> If you have any questions, please leave them here. Thank you.

@HeesungB  I have solved some of the issues in this conversation.

Solved:

1. Removed `@keystonehq/keystone-sdk` from the background and store.
2. Supported light mode.

Need your help:

1. Could you translate the Korean version?

Need discussion:

1. The information icon and interaction were confirmed at the start meeting, why should I remove them?

**Comment by renfengshi** [comment]
@HeesungB The information has been removed.

**Comment by HeesungB** [comment]
> @HeesungB Intl is done. Only one issue is left, which is about the icon files. I cannot understand how to fix.

@renfengshi Could you remove default value of `color` and fix to `fill={color || "currentColor"}` from `fill={color}` in `CameraIcon`, `InformationPlainIcon`?

**Comment by renfengshi** [comment]
@HeesungB All solved. Thank for your help.

**Comment by HeesungB** [comment]
@renfengshi Could you prevent `Claim All` on this component?
Please check below link!

https://github.com/chainapsis/keplr-wallet/blob/d0b311d2382cba6604bdd65c068894617105d5c0/packages/extension/src/pages/main/components/claim-all/index.tsx#L209

File: packages/extension/src/languages/en.json [hunk 14]
```diff
@@ -119,6 +119,25 @@
...
+  "pages.register.connect-keystone.sync": "Sync Keystone",
+  "pages.register.connect-keystone.title-scan": "Scan the QR Code",
+  "pages.register.connect-keystone.paragraph-scan": "Scan the QR code displayed on your Keystone device",
+  "pages.register.connect-keystone.position-qrcode": "Position the QR code in front of your camera.",
+  "pages.register.connect-keystone.no-camera-permission": "No camera permission",
+  "pages.register.connect-keystone.enable-camera": "Please enable your camera permission via [Settings]",
+  "pages.register.connect-keystone.invalid-qrcode": "Invalid QR code",
+  "pages.register.connect-keystone.select-valid-qrcode": "Please ensure you have selected a valid QR code from your Keystone device.",
+  "pages.register.connect-keystone.tutorial": "Tutorial",
+  "pages.register.connect-keystone.note-suggest-chain": "Please note that when adding specific chains to your wallet, the ‘Suggest Chain’ feature may not work.",
```
Comment (HeesungB) [inline packages/extension/src/languages/en.json:137]: @renfengshi Could you change to `Non-Native Chain` from ‘Suggest Chain’
We didn't use ‘Suggest Chain’ anymore

**Comment by renfengshi** [comment]
@HeesungB All solved!

**Comment by renfengshi** [comment]
@HeesungB Solved conflicts.

**Comment by renfengshi** [comment]
If there is any concern about Coin Type, Please contact Keystone.

**Comment by renfengshi** [comment]
If you have any issues with Keystone, please contact us as we are responsible for it.

**Comment by Thunnini** [comment]
#817
